### DISCUSSION TITLE
Wlan0 Interface in Wifibox deosnt exist

### DISCUSSION THREAD

**Discussion by silverhadch** [pgj/freebsd-wifibox#113] [Q&A] [answered]
It simply isnt there but the wifibox works completly except for wifi. Also trying to start it via netif service says not a valid interface lol help.

**Comment by pgj** [comment]
There is no `wlan0` interface in case of using Wifibox.  Well, at least it is not visible on the host.  Instead, you should get a `wifibox0` "virtual" wireless interface, which is actually a bridge for the guest that has the `wlan0` interface and communicates directly with the wireless card.

**Reply by silverhadch** [reply]
No inside the wifibox directly when in console. I simply have no idea how to connect to wifi, i setup wifibox and gave all the info wpa suplicant of the wifibox needed but i cant ifconfig wifibox0 up scan list. Or connect. Help.

**Reply by pgj** [reply]
With the `wifibox console` command you can enter the Linux _guest_ where there should be a `wlan0` interface.  On the contrary, on the FreeBSD _host_, there should be a `wifibox0` interface.  If the `wlan0` interface is missing for the _guest_, there can be many reasons for that.  One typical example of that is that the wireless device was not recognized or the PCI pass-through was not even configured for that.

Would you mind sharing the following:
- Contents of `bhyve.conf` on the _host_
- Output of `dmesg` and `lspci -k` commands from the Wifibox _guest_ (on `wifibox console`)

**Reply by pgj** [reply]
The `wifibox0` interface cannot be used as a wireless interface.  That is, commands like `ifconfig wifibox0 up scan list` will not work.  It should be instead configured as it was a regular network interface.  There is an example of that on the `wifibox(8)` manual page.

**Reply by silverhadch** [reply]
Explain i read the page and am just more confused. ?

**Reply by pgj** [reply]
Could you please share the contents of `/usr/local/etc/wifibox/bhyve.conf` and the output of the `pciconf -lv` on the FreeBSD host?

**Reply by silverhadch** [reply, marked as answer]
no i just decided i dont care anymore

**Reply by pgj** [reply]
All right.
